qemu/tests/x86_cpu_model_boot.py

```diff
@@ -0,0 +1,149 @@
+                <memory>1</memory>
+                <os>
+                    <type arch='%s' machine='%s'>hvm</type>
+                </os>
+                <cpu mode='host-model'/>
+            </domain>
+          """ % (name, vm_arch, machine)
+    xml_file = os.path.join(data_dir.get_tmp_dir(), "temp_xml_for_cpu")
+    with open(xml_file, "w") as f:
+        f.write(xml)
+    try:
+        logging.info("Get cpu model and features from virsh")
+        virsh.define(xml_file)
+        virsh.start(name)
+    except Exception as err:
+        logging.error(err)
+        return
+    else:
+        qemu_cpu_info = cpu.parse_qemu_cpu_flags(
+            cpu.get_cpu_info_from_virsh_qemu_cli(name)['flags'])
+        libvirt_cpu_info = get_cpu_info_from_dumpxml(name)['features']
+        params["qemu_cpu_info"] = qemu_cpu_info
+        params["libvirt_cpu_info"] = libvirt_cpu_info
+
+        qemu_cpu_info = get_flags_patterns(qemu_cpu_info)
+        for flag in libvirt_cpu_info.keys():
+            if flag not in qemu_cpu_info.keys():
+                if libvirt_cpu_info[flag] == 'on':
+                    logging.error("miss some flags from qemu cli")
+                    return False
+            elif libvirt_cpu_info[flag] != qemu_cpu_info[flag]:
+                logging.error("mismatch the option between libvirt and qemu")
+                return False
+        return True
+    finally:
+        if virsh.is_alive(name):
+            virsh.destroy(name, ignore_status=True)
+        virsh.undefine(name, ignore_status=True)
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Test cpu models with full flags that libvirt generates
+    1) start a default configuration xml by virsh
+    2) get the cpu elements from libvirt xml by 'virsh dumpxml $domain'
+    3) parse the cpu_flags and store into libvirt_cpu_flags
+    4) get the cpu flags from qemu process and stroe to qemu_cpu_flags
+    5) Compare compare the cpu flags of xml and qemu process, should be same
+    6) Boot guest with qemu_cpu_flags, guest works well
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    if not compare_cpu_flags(test, params):
+        logging.error("\ncpu libvirt flags are %s\n",
+                      (str(params["libvirt_cpu_info"])))
+        logging.error("\ncpu qemu cli flags are %s\n",
+                      (str(params["qemu_cpu_info"])))
+        test.fail("CPU info is different between dumpxml and qemu cli")
+
+    params["start_vm"] = "yes"
+    vm_name = params['main_vm']
+    env_process.preprocess_vm(test, params, env, vm_name)
+
+    vm = env.get_vm(vm_name)
+    error_context.context("Try to log into guest", logging.info)
+    session = vm.wait_for_login()
+
+    if params.get("reboot_method"):
+        error_context.context("Reboot guest '%s'." % vm.name, logging.info)
+        session = vm.reboot(session=session)
```

Comment:
session is useless in this case.  `vm.reboot()` is enough.

Reply:
> session is useless in this case. `vm.reboot()` is enough.

update and thanks.